lib/galaxy/workflow/scheduling_manager.py

```diff
@@ -54,7 +54,9 @@ def __init__(self, app):
         self.__plugin_classes = self.__plugins_dict()
         self.__init_schedulers()
 
-        if self._is_workflow_handler():
+        if self._is_workflow_handler() or self.app.is_job_handler:
```

Comment:
It's a bit unfortunate we have to run this extra thread indefinitely just for this.

Reply:
It is ... we could also remove this again in 21.09, although this might be nice for admins that move from combined job/workflow handlers to separated handlers. Or we could make it configurable.

Reply:
Configurable seems easy/useful to implement.

Reply:
Actually, things that aren't supposed to be schedulers could just check for any invocations assigned to them at startup and then decide whether or not to start the scheduler? Or reassign the invocations on startup?